src/Integration.Vsix.UnitTests/CFamily/CLangAnalyzerTests.cs

```diff
@@ -96,21 +101,25 @@ public void ExecuteAnalysis_RequestCannotBeCreated_PCH_NoLogOutput()
 
             testSubject.ExecuteAnalysis("path", "charset", new[] { AnalysisLanguage.CFamily }, Mock.Of<IIssueConsumer>(), analysisOptions, CancellationToken.None);
 
+            testSubject.WaitForFactoryCallToComplete();
+
             testLogger.AssertNoOutputMessages();
         }
 
         [TestMethod]
         public void ExecuteAnalysis_RequestCanBeCreated_AnalysisIsTriggered()
         {
             var analysisOptions = new CFamilyAnalyzerOptions();
-            var requestFactory = CreateRequestFactory("path", analysisOptions, Mock.Of<IRequest>());
+            var request = CreateRequest("path");
+            var requestFactory = CreateRequestFactory("path", analysisOptions, request);
+
 
             var testSubject = CreateTestableAnalyzer(
                 requestFactory: requestFactory.Object);
 
-            testSubject.ExecuteAnalysis("path", "charset", new[] { AnalysisLanguage.CFamily }, Mock.Of<IIssueConsumer>(), analysisOptions, CancellationToken.None);
```

Comment:
Previously this method was testing work that was done on the "main" thread before the background thread was started.
Now everything is done on the background thread, so we have to do some extra work to make sure the background thread has finished before we check the results.